### PR TITLE
Feature: moved simplified validation state from prefect-design

### PR DESCRIPTION
**Pull request by stackoverfloweth**
rather than moving in the route of [simplifying prefect-design](https://github.com/PrefectHQ/prefect-design/pull/556) I decided to move the model to `useValidation`.

since prefect-design has vue-compositions as a dependency, it makes sense that prefect-design gets updated to use this model. This enables the best devex

```ts
const { error, state } = useValidation(accountId, 'Account')
```

vs if prefect-design owns the model we need

```ts
const { error, valid, pending, validated } = useValidation(accountId, 'Account')
const state = reactive({ valid, pending, validated })
```

### CODE REVIEW

File: src/useValidation/validationState.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export type ValidationState = {
```
Comment (pleek91) [inline src/useValidation/validationState.ts:1]: Something I've been doing recently is including a reactive of all the refs a composition returns. This is pretty close to that but without a few of the refs. Its a nice devx to be able to work with a reactive vs refs when able and I think it makes sense for this composition to expose everything both as a ref and in this reactive state. 

prefect-design can just use the pieces it needs (I'd have prefect design's `state` still just be the limited set that it actually uses).

Comment (stackoverfloweth) [reply src/useValidation/validationState.ts:1]: can you give me an example of where you do this? I'm curious how to define it's type and what to name the property

Comment (pleek91) [reply src/useValidation/validationState.ts:1]: That's what I would do

File: src/useValidation/useValidation.ts [hunk 2]
```diff
@@ -52,13 +56,22 @@ export function useValidation<T>(
     return valid.value
   }
 
-  const validation: UseValidation = {
+  const state = reactive({
```
Comment (pleek91) [inline src/useValidation/useValidation.ts:59]: One last thought. Would it make more sense for this reactive object to be named `validation`? `state` matches up nicely with prefect-design but it make less sense here as a standalone thing. I don't feel strongly, approving this either way. 

```typescript
const { validation } = useValidation()
```

**Review by pleek91 — APPROVED** [review]
Good enhancement 👍

File: src/useValidation/useValidation.ts (hunk 2 above)
Comment (stackoverfloweth) [reply src/useValidation/useValidation.ts:59]: I think I like `state` more, even without considering it's alignment with how we currently use vee-validate in prefect-design.

**Pull request merged**